src/shared/Core/Trace2.cs

```diff
@@ -286,6 +301,30 @@ internal static bool TryGetPipeName(string eventTarget, out string name)
         return false;
     }
 
+    internal int GetProcessDepth()
+    {
+        if (!_initialized)
+        {
+            return 0;
+        }
+
+        char processSeparator = '/';
+        if (!_sid.Contains(processSeparator))
+        {
+            return 0;
+        }
+
+        int count = 0;
+        // Use AsSpan() for slight performance bump over traditional foreach loop.
+        foreach (var c in _sid.AsSpan())
+        {
+            if (c == processSeparator)
+                count++;
+        }
+
+        return count;
```

Comment:
Could we cache this value, given `_sid` is set once per process? Would avoid rescanning O(N) the `_sid` string each time we called `GetProcessDepth()`.

We could also pre-compute this depth when we do `CreateSid()` too!

Reply:
Now pre-computing with `CreateSid`. Thanks for the idea!